<?xml version="1.0"?>
<launch>

  <!-- change this next line to include the launch file for your robot's hardware drivers: -->

  <!--node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="5.0" />
  </node-->


  <rosparam param="base_width">0.6</rosparam>
  <rosparam param="ticks_meter">1330</rosparam>
  <!--21.3*11*2=468.6   1/(115*3.14)*468.6-->


  
  <node name="l_encoder" pkg="rover_bringup" type="encoder.py" output="screen">
      <remap from="motor_cmd" to="left_wheel_speed"/>
      <remap from="wheel_vtarget" to="lwheel_vtarget"/>
      <remap from="wheel_vel" to="lwheel_vel"/>
      <remap from="wheel" to="lwheel"/>

      <rosparam param="pinA">23</rosparam>
      <rosparam param="pinB">24</rosparam>      
      <rosparam param="Kp">10</rosparam>
      <rosparam param="Ki">0.5</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">-255</rosparam>
      <rosparam param="out_max">255</rosparam>
      <rosparam param="rate">30</rosparam>
      <rosparam param="timeout_ticks">4</rosparam>
      <rosparam param="rolling_pts">2</rosparam>
  	  <!--rosparam param="ticks_meter">1420</rosparam-->
  </node>

   <node name="r_encoder" pkg="rover_bringup" type="encoder.py" output="screen">
      <remap from="motor_cmd" to="right_wheel_speed"/>
      <remap from="wheel_vtarget" to="rwheel_vtarget"/>
      <remap from="wheel_vel" to="rwheel_vel"/>
      <remap from="wheel" to="rwheel"/>

      <rosparam param="pinA">20</rosparam>
      <rosparam param="pinB">21</rosparam>      
      <rosparam param="Kp">10</rosparam>
      <rosparam param="Ki">0.5</rosparam>
      <rosparam param="Kd">0</rosparam>
      <rosparam param="out_min">-255</rosparam>
      <rosparam param="out_max">255</rosparam>
      <rosparam param="rate">30</rosparam>
      <rosparam param="timeout_ticks">4</rosparam>
      <rosparam param="rolling_pts">2</rosparam>
  	  <!--rosparam param="ticks_meter">1420</rosparam-->
  </node>


  <node pkg="rover_bringup" type="twist_to_motors.py" name="twist_to_motors" output="screen"/>
  <node pkg="rover_bringup" type="diff_tf.py" name="diff_tf" output="screen"/> 
  <node pkg="rover_bringup" type="rover_ros.py" name="rover_ros" output="screen"/> 

</launch>
